Added blog frontend and backend

Sourabh782 opened the pull request:
## 📋 Description

<!-- Please provide a brief description of your changes and explain the purpose of the PR. -->

## 🔨 Changes Made

<!-- List the changes made in this PR. If it fixes an issue, please include the issue number (e.g., Fixes #123). -->

- Added blog frontend
- Added blog backend
- 

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [X] My code follows the style guidelines of this project.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings.
- [X] I have tested it locally and it works fine.
- [X] Any dependent changes have been merged and published in downstream modules.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [X] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## 🤝 Related Issues

<!-- List any issues related to this PR. Link them using "Fixes #issue-number" to automatically close them when the PR is merged. -->

- Fixes #37 

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [ ] Test A
- [ ] Test B

## 📸 Screenshots (if applicable)

<!-- If your changes include any UI updates, add screenshots here to show the changes. -->

## Mandatory Tasks

- [X] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

himeshparashar commented:
@sudhanshutech  review needed

sudhanshutech commented on components/blog/Update_Card.tsx:
```diff
@@ -0,0 +1,120 @@
+                            <div className="flex justify-between">
+                                { 
+                                    create ? <div className="flex items-start">
+                                        <button data-modal-target="authentication-modal" onClick={()=>createBlog()} className="block text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800" type="button">
```
too much of re defining styles. Could be reused.

sudhanshutech commented on lib/dbConnect.ts:
```diff
@@ -0,0 +1,26 @@
+import mongoose from "mongoose";
```
rename this root file from lib to config if you would .

Sourabh782 replied:
like this? 
![image](https://github.com/user-attachments/assets/d18cbbb8-66d5-4024-9491-0ab085a9869b)

sudhanshutech commented on app/api/blog/route.ts:
```diff
@@ -0,0 +1,193 @@
+        if(!blog){
+            return Response.json({
+                success: false,
+                message: "blog not found"
```
these could defined at one place as error handler middleware so you don't have redefine everywhere.

sudhanshutech commented on components/blog/BlogCard.tsx:
```diff
@@ -0,0 +1,58 @@
+            body: JSON.stringify(data)
+        })
+
+        console.log(res)
```
remove console logs when testing is done because we don't want to display users responses in console.

Sourabh782 replied:
removed

sudhanshutech commented on components/blog/Update_Card.tsx:
```diff
@@ -0,0 +1,120 @@
+            title: titleVar,
+            body: bodyVar
+        }
+        const response = await fetch("/api/blog", {
```
cors is taken care right?

sudhanshutech commented:
@Sourabh782 please if you would provide some recording of your work.

Sourabh782 commented:
> @Sourabh782 please if you would provide some recording of your work.

Yess sure

sudhanshutech commented:
@Sourabh782 how does this blogs look like to non authorized users?

Sourabh782 commented:
> @Sourabh782 how does this blogs look like to non authorized users?

@sudhanshutech  if user are not logged in they will not be able to see update or delete option its only for owner, i have used conditional rendering for it
![image](https://github.com/user-attachments/assets/4226958e-2cfe-4f91-ba61-449f47fbda4b)

Sourabh782 commented:
> @Sourabh782 how does this blogs look like to non authorized users?

Also if user is not logged in so he wont see Add button on top at blog page
![image](https://github.com/user-attachments/assets/111630ac-d181-4ce1-b3b8-b904e96cca55)

Sourabh782 commented:
@sudhanshutech , @himeshparashar can you check now? implemented some changes
1. removed console.log() statements
2. conditional rendering for users not logged in
